python/cudf/cudf/core/column/column.py

```diff
@@ -1251,12 +1257,12 @@ def corr(self, other: "ColumnBase"):
             f"cannot perform corr with types {self.dtype}, {other.dtype}"
         )
 
-    def nans_to_nulls(self) -> "ColumnBase":
+    def nans_to_nulls(self: T) -> T:
         if self.dtype.kind == "f":
-            col = self.fillna(np.nan)
-            newmask = libcudf.transform.nans_to_nulls(col)
-            self = self.set_mask(newmask)
-        return self
+            newmask = libcudf.transform.nans_to_nulls(self)
+            return self.set_mask(newmask)
```

Comment:
You may have just introduced a bug here, IIRC you the `nans_to_nulls` function doesn't `bitwise and` with the existing mask, so you'd need to do that, or convert `nan` to `null` first to get a proper mask.

Reply:
I don't think I changed the implementation here (from the diff).. Perhaps an orthogonal bug? What's the case that we expect to fail?

Reply:
It looks like the `col = self.fillna(np.nan)` is gone?

Reply:
I'd expect something like:
```
[0., np.nan, None]
```
to fail with the current implementation

Reply:
Connected offline, this was fixed in a previous PR and github is showing an outdated diff.